cloudify_hostpool_plugin/tests/test_tasks.py

```diff
@@ -25,130 +34,192 @@
         self.assertEqual(self.ctx.instance.runtime_properties['key'],
-                         self.keyfile)
-        self.assertEqual(self.ctx.instance.runtime_properties['ip'], self.host)
-        self.assertTrue(os.path.exists(self.keyfile))
-
-    def test_acquire_error_response(self):
-        with requests_mock.mock() as m:
-            m.post('{0}/hosts'.format(self.service_url),
-                   content=json.dumps(self.error_response),
-                   status_code=self.error_code)
-            self.assertRaisesRegexp(
-                NonRecoverableError,
-                'Error: {0} Reason: {1}'.format(
-                    self.error_code,
-                    self.error_response['error']),
-                tasks.acquire,
-                self.service_url,
-                ctx=self.ctx)
-
-    def test_acquire_bad_response(self):
-        with requests_mock.mock() as m:
-            m.post('{0}/hosts'.format(self.service_url),
-                   content="",
-                   status_code=self.error_code,
-                   reason=self.reason)
-            self.assertRaisesRegexp(
-                NonRecoverableError,
-                'Error: {0} Reason: {1}'.format(
-                    self.error_code,
-                    self.reason),
-                tasks.acquire,
-                self.service_url,
-                ctx=self.ctx)
-
-    def test_release_good_response(self):
-        with open(self.keyfile, 'w') as f:
-            f.write(self.key_content)
-        service_url = 'http://test_url'
-        self.ctx.instance.runtime_properties['host_id'] = \
-            self.good_response_acquire_key['host_id']
-        self.ctx.instance.runtime_properties['key'] = self.keyfile
-        with requests_mock.mock() as m:
-            m.delete('{0}/hosts/{1}'.format(service_url, self.host_id),
-                     content=json.dumps(self.good_response_acquire_key),
-                     status_code=requests.codes.OK)
-            tasks.release(service_url, ctx=self.ctx)
-        self.assertFalse(os.path.exists(self.keyfile))
-
-    def test_release_bad_response(self):
-        self.ctx.instance.runtime_properties['host_id'] = \
-            self.good_response_acquire_key['host_id']
-        with requests_mock.mock() as m:
-            m.delete('{0}/hosts/{1}'.format(self.service_url, self.host_id),
-                     content="",
-                     status_code=self.error_code,
-                     reason=self.reason)
-            self.assertRaisesRegexp(
-                NonRecoverableError,
-                'Error: {0} Reason: {1}'.format(
-                    self.error_code,
-                    self.reason),
-                tasks.release,
-                self.service_url,
-                ctx=self.ctx)
+                         self.opts.keyfile)
+        self.assertEqual(self.ctx.instance.runtime_properties['ip'],
+                         self.opts.host)
+        self.assertTrue(os.path.exists(self.opts.keyfile))
```

Comment:
are we certain here that the operation created this file and it's not left over from a previous run. I guess when it runs in travis/circle ci, then it's definitely not left over from a previous run, but I want to make sure, because I didn't see any clean up.